src/main/java/appeng/services/compass/CompassRegion.java

```diff
@@ -85,18 +79,20 @@ private void openFile( boolean create )
 
 	private File getFileName()
 	{
-		String folder = this.rootFolder.getPath() + File.separatorChar + "compass";
-		File folderFile = new File( folder );
+		assert this.rootFolder.isDirectory();
```

Comment:
this is actually just checking the invariant, so adding the PreCondition to the current solution is the way to go